src/components/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -1,216 +1,84 @@
 
-        const format: string = "#,##0.00"; // TODO: will implement with SegmentBy or Color Legend
         const geoDataIndex: IGeoDataIndex = getGeoDataIndex(buckets);
         const data = executionResult.data as Execution.DataValue[][];
         const classes = ["geo-legend", `position-${position}`];
         return (
             <div className={classNames(...classes)}>
-                {(geoDataIndex.hasOwnProperty(COLOR) || geoDataIndex.hasOwnProperty(SEGMENT_BY)) &&
-                    this.renderPushpinColorOrSegmentByLegend(geoDataIndex.hasOwnProperty(SEGMENT_BY))}
-                {geoDataIndex.hasOwnProperty(SIZE) && (
-                    <IntlWrapper locale={locale}>
-                        <IntlTranslationsProvider>
-                            {(props: ITranslationsComponentProps) =>
-                                this.renderPushpinSizeLegend(
-                                    (data[geoDataIndex.size] || []).map(stringToFloat),
-                                    format,
-                                    props.numericSymbols,
-                                )
-                            }
-                        </IntlTranslationsProvider>
-                    </IntlWrapper>
-                )}
+                {geoDataIndex.hasOwnProperty(SIZE) &&
+                    this.renderPushpinSizeLegend(
+                        (data[geoDataIndex.size] || []).map(stringToFloat),
+                        this.getFormatFromExecution(geoDataIndex.size, executionResponse),
+                        locale,
+                    )}
             </div>
         );
     }
 
-    private renderPushpinSizeLegend(sizeValues: number[], format: string, numericSymbols: string[]) {
+    private getFormatFromExecution(indexMeasure: number, result: Execution.IExecutionResponse): string {
+        const DEFAULT_FORMAT = "";
+        if (isEmpty(result)) {
+            return DEFAULT_FORMAT;
+        }
+        const measureGroupHeaderItem = get(result, "dimensions[0].headers[0].measureGroupHeader.items", []);
+        if (measureGroupHeaderItem.length > 0) {
+            return get(measureGroupHeaderItem[indexMeasure], "measureHeaderItem.format", DEFAULT_FORMAT);
+        }
+        return DEFAULT_FORMAT;
+    }
+
+    private renderPushpinSizeLegend(sizeValues: number[], format: string, locale: string) {
         const values: number[] = without(sizeValues, null, undefined, NaN);
         const minValue: number = Math.min(...values);
         const maxValue: number = Math.max(...values);
-        return minValue !== maxValue ? (
-            <PushpinSizeLegend
-                format={format}
-                numericSymbols={numericSymbols}
-                minValue={minValue}
-                maxValue={maxValue}
-            />
-        ) : (
-            ""
+        if (minValue === maxValue) {
+            return "";
+        }
+        return (
+            <IntlWrapper locale={locale}>
+                <IntlTranslationsProvider>
```

Comment:
Could we move these into `PushpinSizeLegend` component?

Reply:
We need to use IntlTranslationsProvider to display format Lable